pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -174,6 +175,10 @@ func (r *Reconciler) install(manifest *mf.Manifest, instance *servingv1alpha1.Kn
 		instance.Status.MarkInstallFailed(err.Error())
 		return err
 	}
+	if !instance.Status.IsInstalled() {
+		duration := time.Since(instance.Status.LastInstallStatusTransitionTime())
```

Comment:
Maybe adding a timer to the if statement above is the best way to derive duration.

If the last three installs were  T1: Success, T2: Fail, T3: Fail
Then the duration here for T3 would represent the time between T3 and T1, right?

Reply:
You will execute this block only when satisfying both
1. install succeeded, otherwise the err would be returned in the previous block
2. install status was not marked success. Because in one install like T1, the status changes as unknown (initialized) -> fail (installing) -> success (installed), and I want to capture the change from fail to success.